includes/Embed_Block.php

```diff
@@ -0,0 +1,181 @@
+	 *
+	 * @return void
+	 */
+	public function init() {
+		// TOOD: Replace with https://cdn.ampproject.org/amp-story-player-v0.js once this updated version of the script is on prod.
+		wp_register_script( 'amp-story-player', 'https://stamp-animation.web.app/dist/amp-story-player.js', [], 'v0', false );
+		wp_register_style( 'amp-story-player', 'https://cdn.ampproject.org/amp-story-player-v0.css', [], 'v0' );
+
+		$asset_file   = WEBSTORIES_PLUGIN_DIR_PATH . 'assets/js/' . self::SCRIPT_HANDLE . '.asset.php';
+		$asset        = is_readable( $asset_file ) ? require $asset_file : [];
+		$dependencies = isset( $asset['dependencies'] ) ? $asset['dependencies'] : [];
+		$version      = isset( $asset['version'] ) ? $asset['version'] : WEBSTORIES_VERSION;
+
+		$dependencies[] = 'amp-story-player';
+
+		wp_register_script(
+			self::SCRIPT_HANDLE,
+			WEBSTORIES_PLUGIN_DIR_URL . 'assets/js/' . self::SCRIPT_HANDLE . '.js',
+			$dependencies,
+			$version,
+			false
+		);
+
+		wp_register_style(
+			self::STYLE_HANDLE,
+			WEBSTORIES_PLUGIN_DIR_URL . 'assets/css/' . self::STYLE_HANDLE . '.css',
+			[ 'amp-story-player' ],
+			$version
+		);
+
+		// todo: use register_block_type_from_metadata() once generally available.
+
+		// Note: does not use 'script' and 'style' args, and instead uses 'render_callback'
+		// to enqueue these assets only when needed.
+		register_block_type(
+			'web-stories/embed',
+			[
+				'attributes'      => [
+					'url'    => [
+						'type' => 'url',
+					],
+					'title'  => [
+						'type' => 'string',
+					],
+					'poster' => [
+						'type' => 'string',
+					],
+					'width'  => [
+						'type'    => 'number',
+						'default' => 360,
+					],
+					'height' => [
+						'type'    => 'number',
+						'default' => 600,
+					],
+					'align'  => [
+						'type'    => 'string',
+						'default' => 'none',
+					],
+				],
+				'render_callback' => [ $this, 'render_block' ],
+				'editor_script'   => self::SCRIPT_HANDLE,
+				'editor_style'    => self::STYLE_HANDLE,
+			]
+		);
+
+		add_filter( 'wp_kses_allowed_html', [ __CLASS__, 'filter_kses_allowed_html' ], 10, 2 );
+	}
+
+	/**
+	 * Filter the allowed tags for KSES to allow for amp-story children.
+	 *
+	 * @param array|string $allowed_tags Allowed tags.
+	 *
+	 * @return array|string Allowed tags.
+	 */
+	public static function filter_kses_allowed_html( $allowed_tags ) {
+		if ( ! is_array( $allowed_tags ) ) {
+			return $allowed_tags;
+		}
+
+		$story_player_components = [
+			'amp-story-player' => [],
+		];
+
+		$allowed_tags = array_merge( $allowed_tags, $story_player_components );
+
+		return $allowed_tags;
+	}
+
+	/**
+	 * Renders the block type output for given attributes.
+	 *
+	 * @param array  $attributes Block attributes.
+	 * @param string $content    Block content.
+	 *
+	 * @return string Rendered block type output.
+	 */
+	public function render_block( array $attributes, $content ) {
+		// The only 2 mandatory attributes.
```

Comment:
Should we have a different render for `is_feed()`? Maybe something that doesn't require js at all.

Reply:
Good idea. Implemented now.